native-java-gemini-function-calling-example/README.md

```diff
@@ -0,0 +1,187 @@
+# Build Native Java Images with GraalVM with Langchain4J and Gemini Function Calling
+
+Demonstrate `Function Calling` code using Gemini with Langchain4j
+
+__Use Case__: Suppose we want the AI model to respond with information that it does not have.
+For example the status of your recent payment transactions.
+Users can ask questions about current status for certain payment transactions and use function calling to answer them.
+
+__Environment__:
+Please set the following environment variables before running this example:
+```shell
+export VERTEX_AI_GEMINI_PROJECT_ID=<your project id>
+export VERTEX_AI_GEMINI_LOCATION=<region, ex us-central1>
+export VERTEX_AI_GEMINI_MODEL=<the model in use, ex.gemini-1.5-flash-001>
+```
+
+For example, let's consider a sample dataset and a function that retrieves the payment status given a transaction:
+
+```java
+record Transaction(String id) {
+}
+
+record Status(String name) {
+}
+
+private static final Map<Transaction, Status> DATASET =
+	Map.of(
+		new Transaction("001"), new Status("pending"),
+		new Transaction("002"), new Status("approved"),
+		new Transaction("003"), new Status("rejected"));
+
+@Bean
+@Description("Get the status of a payment transaction")
+public Function<Transaction, Status> paymentStatus() {
+	return transaction -> DATASET.get(transaction);
+}
+```
+
+Function is registered as `@Tool`, which are Java methods the language model can use to call. 
+Langchain4j greatly simplifies code you need to write to support function invocation.
+It brokers the function invocation conversation for you.
+
+Lets add the boot starters for 4 AI Models that support function calling:
+
+```xml
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-spring-boot-starter</artifactId>
+    <version>${langchain4j.version}</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-vertex-ai-gemini</artifactId>
+    <version>${langchain4j.version}</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j</artifactId>
+    <version>${langchain4j.version}</version>
+</dependency>
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-core</artifactId>
```

Comment:
minor: this dependency is not required, it is imported transitively with all other dependencies